public/app/features/canvas/elements/ellipse.tsx

```diff
@@ -194,3 +152,23 @@ export const ellipseItem: CanvasElementItem<EllipseConfig, EllipseData> = {
       });
   },
 };
+
+const getStyles = (theme: GrafanaTheme2, data: CanvasElementData | undefined) => ({
+  container: css({
+    height: '100%',
+    width: '100%',
+  }),
+  text: css({
+    position: 'absolute',
+    top: data?.valign === VAlign.Middle ? '50%' : data?.valign === VAlign.Top ? '10%' : '90%',
```

Comment:
Probably shouldn't repeat the logic for this either 😄

Reply:
agree, let me move it outside